stingray/powerspectrum.py

```diff
@@ -582,8 +582,8 @@ def _make_matrix(self, lc):
         self.dyn_ps = np.array([ps.power for ps in ps_all]).T
 
         self.freq = ps_all[0].freq
-        self.time = np.arange(lc.time[0] - 0.5 * lc.dt + 0.5 * self.segment_size,
-                              lc.time[-1] + 0.5 * lc.dt, self.segment_size)
```

Comment:
`lc.time` contains the half intervals, so here it was aligned to the start of the interval before adding `0.5 * segment_size`

Reply:
That may be true, but you still end up with a `time` array that has a different number of bins as the `dyn_ps` attribute. 

Motivating example:
```
tstart = 100.0

ts = 1600.0
dt = 0.01

ttest = np.linspace(tstart, tstart+ts, int(ts/dt))

print(np.diff(ttest))

counts1 = np.random.poisson(100, size=ttest.shape[0])
counts2 = np.random.poisson(100, size=ttest.shape[0])

lc1 = Lightcurve(ttest, counts1, dt=0.01)
lc2 = Lightcurve(ttest, counts2, dt=0.01)

ps_avg = AveragedPowerspectrum(lc=lc1, segment_size=1.0)
cs_avg = AveragedCrossspectrum(lc1=lc1, lc2=lc2, segment_size=1.0)

# in master, the behaviour of AveragedCrossspectrum and AveragedPowerspectrum is different
# but it shouldn't be!
print(ps_avg.m)
print(cs_avg.m)

dps = DynamicalPowerspectrum(lc1, segment_size=1.0)

# in master, the first prints out 1616, the second 1599 elements
print(dps.dyn_ps.shape[1])
print(dps.time.shape)
```

Reply:
Hm. It's a weird difference of length O_o
However, it looks to me that the problem here would be solved in the cleanest way by simply using the same times used to calculate `ps_all`. From line 483:
```
start_inds, end_inds = \
        bin_intervals_from_gtis(self.gti, segment_size, lc.time, dt=lc.dt)
```
and then 
```
for start_ind, end_ind in zip(start_inds, end_inds):
    time = lc.time[start_ind:end_ind]
(...)
```
We could use the mean of each time interval in the `self.time` array. 
This takes into account the GTIs and should yield the exact same times. Looking at the current implementation, I don't even think that the times really are consistent. 
Does it look right to you?

Reply:
yeah, that looks right. Check the newest commit?

Reply:
Looks good!